### PR TITLE
Basic cleanup of sep theory

### PR DESCRIPTION
**Pull request by ajreynol**
Most of the simplifications are due to the fact that we only handle a single heap type. (The solver was initially designed to potentially handle more than one heap type).

### CODE REVIEW

File: src/theory/sep/theory_sep.cpp [hunk 1]
```diff
@@ -1060,33 +1027,35 @@ int TheorySep::processAssertion(
...
+          == d_type_references.end())
+      {
+        d_type_references.push_back(r);
       }
     }
     if( add ){
       //add max cardinality
-      if( card>(int)d_card_max[tn] ){
-        d_card_max[tn] = card;
+      if (card > (int)d_card_max)
```
Comment (mudathirmahgoub) [inline src/theory/sep/theory_sep.cpp:1056]: Can't card be declared as unsigned as well?

File: src/theory/sep/theory_sep.cpp [hunk 2]
```diff
@@ -184,76 +171,95 @@ void TheorySep::computeCareGraph() {
...
+      pto_children.push_back(l);
+      Trace("sep-model") << " " << l << " -> ";
+      if (d_pto_model[l].isNull())
+      {
+        Trace("sep-model") << "_";
+        TypeEnumerator te_range(d_type_data);
+        if (d_env.isFiniteType(d_type_data))
+        {
+          pto_children.push_back(*te_range);
         }else{
```
Comment (mudathirmahgoub) [inline src/theory/sep/theory_sep.cpp:209]: Since most of the file has changed, I prefer to format the whole file, not only the changes.

**Review by mudathirmahgoub — APPROVED** [review]
LGTM

File: src/theory/sep/theory_sep.cpp (hunk 1 above)
Comment (ajreynol) [reply src/theory/sep/theory_sep.cpp:1056]: Sure, I'll refactor.

File: src/theory/sep/theory_sep.cpp (hunk 2 above)
Comment (ajreynol) [reply src/theory/sep/theory_sep.cpp:209]: This would take a lot more effort, I'll do this separately.

**Pull request merged**